docs/concept/#894-Efficient-orchestration-of-edc-negotiations/#894-Efficient-orchestration-of-edc-negotiations.md

```diff
@@ -0,0 +1,103 @@
+# #894 Efficient orchestration of edc negotiations
+
+| Key           | Value                                                                    |
+|---------------|--------------------------------------------------------------------------|
+| Creation date | 09.10.2024                                                               |
+| Ticket Id     | https://github.com/eclipse-tractusx/item-relationship-service/issues/894 |    
+| State         | DONE                                                                     | 
+
+## Table of Contents
+
+1. [Decision](#Decision)
+2. [Rationale](#Rationale)
+3. [Approach](#Approach)
+4. [Testing Strategy](#Testing-Strategy)
+
+## Decision
+
+Implementation of a new EDC Orchestrator within the EDC Client Library for efficient management of negotiations. The
+goal is to prevent duplicate negotiations for assets, thereby ensuring more efficient processing in asset consumption.
+
+## Rationale
+
+Each job is executed in isolation. This results in many jobs conducting a multitude of unnecessary contract negotiations
+for the same DT Registry Asset in the EDC catalog. In case the same catalog offer is requested in different jobs this
+leads to not performant overall executions. Therefore, a concept is required how to handle this in the most efficient
+way.
+
+## Approach
+
+A new service “NegotiationOrchestrator” is introduced which is a middle layer between the already existing components
+EdcSubmodelClientImpl and ContractNegotiationService, EDCCatalogFacade and EndpointDataReferenceCacheService.
+
+The service handles the following tasks:
+
+- orchestration of catalog requests by id
+- orchestration of catalog requests by type
+- orchestration of contract negotiations
+- caching of ongoing contract negotiations
+- caching of EDR tokens
+
+To prevent too many parallel requests to the EDC, an ExecutorService handles the catalog requests and negotiations with
+a defined thread pool.
+
+To cache the currently ongoing negotiations, a ConcurrentHashMap with EDC AssetIds as key can be used. Due to the
```

Comment:
I would recommend describing that in more generic way. I would recommend a caching api with a in the MVP state initial implementation using ConcurrentHashMap and could be later changed to a more robust cache implementation.